api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -19,16 +19,118 @@
+   *
+   * @param path The string representation of the path of the generic file.
+   * @return {@code true}, if the generic file exists; {@code false}, otherwise.
+   * @throws InvalidPathException     If the specified path's string representation is not valid, according to
+   *                                  {@link GenericFilePath#parseRequired(String)}.
+   * @throws AccessControlException   If the user of the current session does not have permission to check the existence
+   *                                  of the specified file.
+   * @throws OperationFailedException If the operation fails for any other (checked) reason.
+   */
+  default boolean doesFileExist( @NonNull String path ) throws OperationFailedException {
+    return doesFileExist( GenericFilePath.parseRequired( path ) );
+  }
 
-  boolean doesFolderExist( @NonNull String path ) throws OperationFailedException;
+  /**
+   * Creates a folder given its path.
+   * <p>
+   * This method ensures that each ancestor folder of the specified folder exists,
+   * creating it if necessary, and allowed.
+   * <p>
+   * When the operation is successful, the folder session cache for the generic file provider owning the folder is
+   * automatically cleared.
+   *
+   * @param path The path of the generic folder to create.
+   * @return {@code true}, if the folder did not exist and was created; {@code false}, if the folder already existed.
+   * @throws AccessControlException   If the user of the current session does not have permission to create the folder.
+   * @throws InvalidPathException     If the folder's path is not valid.
+   * @throws OperationFailedException If the operation fails for any other (checked) reason.
+   * @see #clearFolderCache()
+   */
+  boolean createFolder( @NonNull GenericFilePath path ) throws OperationFailedException;
 
-  boolean createFolder( @NonNull String path ) throws OperationFailedException;
+  /**
+   * Creates a folder given its path's string representation.
+   * <p>
+   * This method ensures that each ancestor folder of the specified folder exists,
+   * creating it if necessary, and allowed.
+   * <p>
+   * The default implementation of this method parses the given path's string representation using
+   * {@link GenericFilePath#parseRequired(String)} and then calls {@link #createFolder(GenericFilePath)} with the result.
+   * <p>
+   * When the operation is successful, the folder tree session cache for the generic file provider owning the folder is
+   * automatically cleared.
+   *
+   * @param path The string representation of the path of the generic folder to create.
+   * @return {@code true}, if the folder did not exist and was created; {@code false}, if the folder already existed.
+   * @throws AccessControlException   If the user of the current session does not have permission to create the folder.
+   * @throws InvalidPathException     If the folder's path is not valid.
```

Comment:
This one makes sense to me because InvalidPathException extends OperationFailedException, but AccessControlException extends SecurityException, which extends from RuntimeException?